web/src/app/add-category/add-category.component.ts

```diff
@@ -53,19 +54,20 @@ export class AddCategoryComponent implements OnInit {
   }
 
   add(event: MatChipInputEvent): void {
-    console.log("add");
-    let name = event.value.trim();
-    if (!name) {
-      return;
-    }
-    let category = this.findExistingCategory(name);
-    if (!category) {
-      this.persistCategory(name)
-    } else {
-      this.addCategory(category);
-    }
+    if (!this.matAutocomplete.isOpen) {
```

Comment:
What if autocomplete is open? Why do we enter here? Looks like we shouldn't enter here in this case.

Reply:
Sorry for the delay. Let's get straight to the point:
1. We need to set `[matChipInputAddOnBlur]="false"` to TRUE in order to be able to save the chips when we click elsewhere, including "save" button or item from the autocomplete list.
2. Since `[matChipInputAddOnBlur]="true"` saves the chip you typed even if you click item from autocomplete list, for instance:
You typed "mush" and clicked to choose "mushrooms" from the autocomplete list - it saves "mush" instead.
You have to add `if (!this.matAutocomplete.isOpen) {` condition to make it right.

Reply:
What if you make [matChipInputAddOnBlur]=matAutocomplete.isOpen?

Reply:
You might mean ` [matChipInputAddOnBlur]="!matAutocomplete.isOpen"`
It works the same way as adding `if (!this.matAutocomplete.isOpen) {` condition.

Reply:
The question is not about "if it works", but "how easy it can be changed".
And from this perspective, there is a big difference. 
In the option when we check it inside the method we break encapsulation and create hard coupling between these two methods.

Reply:
Then it's better not to break encapsulation and write it as `[matChipInputAddOnBlur]="!matAutocomplete.isOpen"`